llvm/lib/Target/Mips/MCTargetDesc/MipsTargetStreamer.cpp

```diff
@@ -1255,7 +1255,9 @@ void MipsTargetELFStreamer::emitDirectiveCpsetup(unsigned RegNo,
     emitRRI(Mips::SD, GPReg, Mips::SP, RegOrOffset, SMLoc(), &STI);
   }
 
-  if (getABI().IsN32()) {
+#if 0
+  // We haven't support -mabicalls -mno-shared yet.
+  if (-mno-shared) {
```

Comment:
IIUC -mno-shared only makes sense for -fno-pic. Modern executables use PIE and cannot use -mno-shared.